Initial commit to data-binding

karlastone4 opened the pull request:
Ways to Improve the program?

karlastone4 committed:
Initial commit to data-binding

wildbiotiger commented on Pirates/Pirates/MainViewModel.cs:
```diff
@@ -0,0 +1,42 @@
+
+            Pirates = new ObservableCollection<PirateNames>
+            {
+
```
remove extra whitespace

wildbiotiger commented on Pirates/Pirates/MainViewModel.cs:
```diff
@@ -0,0 +1,42 @@
+
+namespace Testing2
+{
+   public class MainViewModel
```
add intellisense comments

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,99 @@
+﻿using System;
```
remove unused usings

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,99 @@
+
+        private void EnterAdd(object sender, KeyEventArgs e)
+        {
+
```
remove extra whitespace

wildbiotiger commented on Pirates/Pirates/MainViewModel.cs:
```diff
@@ -0,0 +1,42 @@
+using System.Collections.ObjectModel;
+using System.Windows.Controls;
+
+namespace Testing2
```
rename namespace in all files to something useful
see examples of C# namespace definitions on the web

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,99 @@
+
+        }
+
+
```
remove whitespace

wildbiotiger commented on Pirates/Pirates/ViewModel/PirateNames.cs:
```diff
@@ -0,0 +1,15 @@
+
+namespace Pirates
+{
+    public class PirateNames
```
make this singular

wildbiotiger commented on Pirates/Pirates/ViewModel/Pirates.cs:
```diff
@@ -0,0 +1,12 @@
+
+namespace Pirates.ViewModel
+{
+    class Pirates 
```
make this singular

wildbiotiger commented on VisualStudio.gitignore:
```diff
@@ -0,0 +1,349 @@
+## Ignore Visual Studio temporary files, build results, and
```
remove this file

karlastone4 committed:
Delete VisualStudio.gitignore

karlastone4 committed:
Update MainWindow.xaml.cs

Remove extra whitespace
Remove unused usings
Add comments
Make names signular

karlastone4 committed:
Update MainWindow.xaml

Make names singular

karlastone4 committed:
Update Pirate.cs

Update namespace to Pirate

karlastone4 committed:
Update App.xaml

Change names to singular

karlastone4 committed:
Update App.xaml.cs

Make names singular

karlastone4 committed:
Update Pirate.cs

Make class singular

karlastone4 committed:
Update PirateNames.cs

Remove empty whitespaces
Make class PirateName

karlastone4 committed:
Delete MainViewModel.cs

karlastone4 committed:
Delete Pirate.cs

karlastone4 committed:
Update MainViewModel.cs

Remove extra white space
Add comments to explain methods
Change collection to singular name
Remove unused usings

karlastone4 reviewed (commented):
Updated files

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,47 @@
+        public MainWindow()
+        {
+            InitializeComponent();
+        }
```
add single spacing between functions

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,47 @@
+        {
+            InitializeComponent();
+        }
+        //adding the textbox info to pirates collection by clicking add button
```
dont use present progressive tense in comments
try "adds the textbox info..."

wildbiotiger commented on Pirates/Pirates/ViewModel/MainViewModel.cs:
```diff
@@ -0,0 +1,30 @@
+
+namespace Pirates
+{
+    public class MainViewModel
```
needs comments

wildbiotiger commented on Pirates/Pirates/ViewModel/Pirates.cs:
```diff
@@ -0,0 +1,12 @@
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pirate.ViewModel
```
needs comments

karlastone4 committed:
Update data-binding

karlastone4 committed:
Update data-binding

Make comments more concise

karlastone4 committed:
Delete MainViewModel.cs

karlastone4 committed:
Delete 2nd Main View Model file

karlastone4 committed:
Merge branch 'kstone/data-binding' of https://github.com/wildbiotiger/dotnet-intro-kstone into kstone/data-binding

karlastone4 committed:
Update MainViewModel

Make comments more concise

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml:
```diff
@@ -0,0 +1,39 @@
+            <RowDefinition Height="auto" MinHeight="73"  />
+            <RowDefinition Height="30*"/>
+        </Grid.RowDefinitions>
+        <TextBox x:Name="TextBox" Grid.ColumnSpan="2" Margin="10,16,21,5" TextWrapping="Wrap" KeyDown="EnterAdd" KeyUp="EnterClear"/>
```
remove unused names
sort the attributes so that grid.XXX is first and margins are second
events and interaction should be last
space out the attributes on multiple lines so they dont go offscreen

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,55 @@
+            InitializeComponent();
+        }
+
+        //add the textbox info to pirates collection by clicking add button
```
turn these into intellisense comments

wildbiotiger commented on Pirates/Pirates/Pirate.cs:
```diff
@@ -0,0 +1,12 @@
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Testing2.Directory
```
rename

wildbiotiger commented on Pirates/Pirates/ViewModel/MainViewModel.cs:
```diff
@@ -0,0 +1,36 @@
+        public static MainViewModel DataContext { get; set; }
+        public ObservableCollection<PirateName> Pirates { get; set; }
+
+        //new method for shorthand adding pirate names from collection
```
convert to intellisense comments

karlastone4 committed:
Update data-binding

Make normal comments intellisense comments
Edit xaml code to look nicer

karlastone4 committed:
Delete Pirate.cs

karlastone4 committed:
Update MainIVewModel

make comments intellisense comments

karlastone4 committed:
Merge branch 'kstone/data-binding' of https://github.com/wildbiotiger/dotnet-intro-kstone into kstone/data-binding

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml:
```diff
@@ -22,10 +22,14 @@
+                VerticalAlignment="Bottom" Click="Add_Click"/>
+        <Button Margin="5,0,10,7" Grid.Column="1" x:Name="Remove" Content="Remove"   Background="#FFDD4F4F" 
+                Grid.Row="1" VerticalAlignment="Bottom" Click="Remove_Click"/>
+        <ListBox Grid.ColumnSpan="2" Margin="10,10,10,16" Grid.Row="2" Grid.RowSpan="2" x:Name="ListBox"  
```
move grid properties before margin

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml.cs:
```diff
@@ -8,41 +8,41 @@ namespace Pirate
     /// </summary>
     
-    // class for Mainwindow code 
+    /// class for Mainwindow code 
```
make these intellisense comments, three slashes is not the same as the XML syntax intellisense produces

wildbiotiger commented on Pirates/Pirates/ViewModel/Pirates.cs:
```diff
@@ -4,11 +4,11 @@
+/// <summary>
+/// Class for ViewModel
+/// </summary>
 namespace Pirate.ViewModel
```
the namespace does not need comments, make it above the class

karlastone4 committed:
Update Pirates

Utilize App.xaml to chagne UI

wildbiotiger commented on .gitignore:
```diff
@@ -337,4 +337,6 @@ ASALocalRun/
\ No newline at end of file
+healthchecksdb
+Pirates/Pirates/BlackPearl.jpg
+Pirates/Pirates/ViewModel/BlackPearl.png
```
why are these being ignored?

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml:
```diff
@@ -7,38 +7,51 @@
+            <RowDefinition Height="52"/>
+            <RowDefinition Height="60*"/>
+            <RowDefinition Height="60*"/>
+            <RowDefinition Height="60*" />
```
for both rows and cols, reduce the star parameters and use auto for everything else

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml:
```diff
@@ -7,38 +7,51 @@
-                   Text="Enter Text Here" VerticalAlignment="Top" Width="96"/>
+        <Label Grid.Row="5" Grid.ColumnSpan="4" Style="{StaticResource Displays}" x:Name="NameBlock"/>
+        <Label  Grid.ColumnSpan="4"  Grid.Row="6" x:Name="ShipBlock" Style="{StaticResource Displays}" />
+        <Label Grid.ColumnSpan="4" Grid.Row="7"  x:Name="PortBlock" Style="{StaticResource Displays}"/>
```
remove x:Names for the items that dont need it

wildbiotiger commented on Pirates/Pirates/MainWindow.xaml.cs:
```diff
@@ -8,48 +9,89 @@ namespace Pirate
             InitializeComponent();
         }
 
         ///add the textbox info to pirates collection by clicking add button
```
change this to a full intellisense comment - do this for all other comments

wildbiotiger commented on Pirates/Pirates/ViewModel/MainViewModel.cs:
```diff
@@ -9,28 +10,55 @@ public class MainViewModel
-    }
 
+        ///method to remove ship from collection
+        public void RemoveShip(int loseShip)
```
is this function removing a ship or a pirate? seems redundant

karlastone4 committed:
Update Pirates

Make comments actual Intellisense comments
Change row and column sizes
Remove usings that are not needed

karlastone4 committed:
Merge branch 'kstone/data-binding' of https://github.com/wildbiotiger/dotnet-intro-kstone into kstone/data-binding

wildbiotiger commented on Pirates/Pirates/ViewModel/MainViewModel.cs:
```diff
@@ -24,41 +24,52 @@ public MainViewModel()
-         public void AddPerson(string newPerson)
+        /// </summary>
+        /// <param name="newPerson"></param>
+        public void AddPerson(string newPerson)
```
remove the addperson/ship/landing functions, those aren't quite what we need them to be. they don't accomplish what you think they do.

karlastone4 committed:
Update Pirates

Delete unused methods

karlastone4 committed:
Update Pirate

Remove unused code
